src/CMakeLists.txt

```diff
@@ -5,20 +5,14 @@ include(FetchContent)
 add_library(aer SHARED aer.hpp generator.hpp)
 target_include_directories(aer INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
 set_target_properties(aer PROPERTIES LINKER_LANGUAGE CXX)
-
-# Ensure we have proper compile definitions when importing torch
-if (CMAKE_CUDA_COMPILER) 
-  set(CMAKE_CXX_STANDARD 17)
-  set(CMAKE_CUDA_STANDARD 17)
-  set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
-  set(module_compile_definitions USE_CUDA)
-endif()
-
-# Attempt to import Torch
-message(WARNING "USE " ${USE_TORCH})
-if (USE_TORCH)
-  find_package(Torch REQUIRED)
-  set(module_compile_definitions USE_TORCH)
+# set coroutine flags for clang appropriately
+# thanks to https://stackoverflow.com/questions/64703866/is-clang-11s-invocation-of-coroutine-headers-via-fcoroutines-flag
```

Comment:
If this is well organised it will be easier to debug and extend in the future

https://stackoverflow.com/questions/45955272/modern-way-to-set-compiler-flags-in-cross-platform-cmake-project
https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html
https://cmake.org/cmake/help/v3.3/command/add_compile_options.html

Reply:
Good point. I'll definitely change this to the add_compile_options thing you suggested.
But I'm also happy to improve on this going forward. C.f. the discussion about meson, for instance.